File: ABFRealmTableViewControllerExample/ABFRealmTableViewControllerExample/TaskObject.m
//
//  TaskObject.m
//  ABFRealmTableViewControllerExample
//
//  Created by Adam Fish on 8/5/15.
//  Copyright (c) 2015 Adam Fish. All rights reserved.
//

#import "TaskObject.h"

@implementation TaskObject

// Specify default values for properties

+ (NSDictionary *)defaultPropertyValues
{
    return @{@"taskDescription": @""};
}

+ (NSString *)primaryKey
{
    return @"taskId";
}

+ (instancetype)taskWithDescription:(NSString *)description
{
    NSUInteger taskCount = [TaskObject allObjects].count;
    
    TaskObject *task = [[TaskObject alloc] init];
    task.taskId = taskCount;
    task.taskDescription = description;
    
    return task;
}

@end

File: SwiftExample/RealmTableViewControllerExample/TaskObject.swift
//
//  TaskObject.swift
//  RealmTableViewControllerExample
//
//  Created by Adam Fish on 10/7/15.
//  Copyright © 2015 Adam Fish. All rights reserved.
//

import RealmSwift

class TaskObject: Object {
    // MARK: Model
    dynamic var taskId = 0
    dynamic var taskDescription = ""
    
    // MARK: Realm Swift
    override static func primaryKey() -> String? {
        return "taskId"
    }
    
    class func task(description: String) -> TaskObject {
        let task = TaskObject()
        
        task.taskId = try! Realm().objects(TaskObject).count
        task.taskDescription = description
        
        return task
    }
}
